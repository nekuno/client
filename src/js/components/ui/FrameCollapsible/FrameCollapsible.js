import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Frame from '../Frame/Frame.js';
import styles from './FrameCollapsible.scss';

export default class FrameCollapsible extends Component {

    static propTypes = {
        title         : PropTypes.string
    };

    constructor(props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);

        this.state = {
            open: false
        };
    }

    handleClick() {
        const {open} = this.state;

        this.setState({open: !open});
    }

    render() {
        const {title, children} = this.props;
        const {open} = this.state;

        return (
            <Frame>
                <div className={styles.frameCollapsible}>
                    <div className={styles.frameTop} onClick={this.handleClick}>
                        <div className={styles.title}>
                            {title}
                        </div>
                        <div className={styles.arrow}>
                            {open ?
                                <div className={styles.arrow + ' icon icon-chevron-up'}/>
                                :
                                <div className={styles.arrow + ' icon icon-chevron-down'}/>
                            }
                        </div>
                    </div>
                    {open ?
                        <div className={styles.content}>
                            {children}
                        </div>
                        : null
                    }
                </div>
            </Frame>
        );
    }
}