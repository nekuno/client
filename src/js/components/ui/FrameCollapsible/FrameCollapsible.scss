@import '../../../../scss/_partials/colors';

.frame-collapsible {

  .frame-top {
    cursor: pointer;
    font-size: 18px;
    width: 100%;

    .title {
      font-weight: 600;
      display: inline-block;
      padding: 3px 3%;
      width: 69%;
    }
    .arrow {
      display: inline-block;
      font-weight: bold;
      padding: 3px 3%;
      text-align: right;
      width: 19%;
    }
  }
  .content {
    margin: 24px 0 8px 3%;
  }
}