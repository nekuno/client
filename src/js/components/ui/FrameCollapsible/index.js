import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Frame from '../Frame/';
import styles from './FrameCollapsible.scss';
import IconCollapsible from "../IconCollapsible";

export default class FrameCollapsible extends Component {

    static propTypes = {
        title         : PropTypes.string,
        onToggle : PropTypes.func,
    };

    constructor(props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);

        this.state = {
            open: false
        };
    }

    handleClick() {
        const {open} = this.state;

        this.setState({open: !open});

        this.props.onToggle();
    }

    render() {
        const {title, children} = this.props;
        const {open} = this.state;

        return (
            <Frame>
                <div className={styles.frameCollapsible}>
                    <div className={styles.frameTop} onClick={this.handleClick}>
                        <div className={styles.title}>
                            {title}
                        </div>
                        <div className={styles.arrow}>
                            <IconCollapsible open={open}/>
                        </div>
                    </div>
                    <div className={styles.content} style={{display: open ? 'block' : 'none'}}>
                        {children}
                    </div>
                </div>
            </Frame>
        );
    }
}

FrameCollapsible.defaultProps = {
    onToggle: () => {},
};