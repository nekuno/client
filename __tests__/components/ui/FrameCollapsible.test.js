import React from 'react';
import FrameCollapsible from '../../../src/js/components/ui/FrameCollapsible/FrameCollapsible.js';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test FrameCollapsible component', () => {
    const frameCollapsible = shallow(<FrameCollapsible title={'Title'}>Lorem ipsum</FrameCollapsible>);

    it('should be defined', () => {
        expect(FrameCollapsible).toBeDefined();
    });
    it('should render correctly', () => {
        expect(frameCollapsible).toMatchSnapshot();
    });
});
