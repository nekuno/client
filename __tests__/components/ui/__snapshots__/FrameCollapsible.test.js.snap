// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FrameCollapsible component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FrameCollapsible
    title="Title"
  >
    Lorem ipsum
  </FrameCollapsible>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="frameCollapsible"
      >
        <div
          className="frameTop"
          onClick={[Function]}
        >
          <div
            className="title"
          >
            Title
          </div>
          <div
            className="arrow"
          >
            <div
              className="arrow icon icon-chevron-down"
            />
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="frameTop"
            onClick={[Function]}
          >
            <div
              className="title"
            >
              Title
            </div>
            <div
              className="arrow"
            >
              <div
                className="arrow icon icon-chevron-down"
              />
            </div>
          </div>,
          null,
        ],
        "className": "frameCollapsible",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
              >
                Title
              </div>,
              <div
                className="arrow"
              >
                <div
                  className="arrow icon icon-chevron-down"
                />
              </div>,
            ],
            "className": "frameTop",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Title",
                "className": "title",
              },
              "ref": null,
              "rendered": "Title",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="arrow icon icon-chevron-down"
                />,
                "className": "arrow",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "arrow icon icon-chevron-down",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="frameCollapsible"
        >
          <div
            className="frameTop"
            onClick={[Function]}
          >
            <div
              className="title"
            >
              Title
            </div>
            <div
              className="arrow"
            >
              <div
                className="arrow icon icon-chevron-down"
              />
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="frameTop"
              onClick={[Function]}
            >
              <div
                className="title"
              >
                Title
              </div>
              <div
                className="arrow"
              >
                <div
                  className="arrow icon icon-chevron-down"
                />
              </div>
            </div>,
            null,
          ],
          "className": "frameCollapsible",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
                >
                  Title
                </div>,
                <div
                  className="arrow"
                >
                  <div
                    className="arrow icon icon-chevron-down"
                  />
                </div>,
              ],
              "className": "frameTop",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Title",
                  "className": "title",
                },
                "ref": null,
                "rendered": "Title",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="arrow icon icon-chevron-down"
                  />,
                  "className": "arrow",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "arrow icon icon-chevron-down",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          null,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
